packages/store-sync/src/createStoreSync.ts

```diff
@@ -263,7 +268,9 @@ export async function createStoreSync<config extends StoreConfig = StoreConfig>(
     // This currently blocks for async call on each block processed
     // We could potentially speed this up a tiny bit by racing to see if 1) tx exists in processed block or 2) fetch tx receipt for latest block processed
     const hasTransaction$ = recentBlocks$.pipe(
-      concatMap(async (blocks) => {
+      // We use `mergeMap` instead of `concatMap` here to send the fetch request immediately when a new block range appears,
+      // instead of sending the next request only when the previous one completed.
+      mergeMap(async (blocks) => {
```

Comment:
can we separate out these changes into its own PR?

Reply:
pulling out into a separate PR

Reply:
Followup here: https://github.com/latticexyz/mud/pull/2665